runtime/Java/src/org/antlr/v4/runtime/Parser.java

```diff
@@ -249,6 +249,15 @@ public Token matchWildcard() throws RecognitionException {
 		return t;
 	}
 
+	protected ParserRuleContext createContext(int ruleIndex, int altIndex, ParserRuleContext parent, int invokingState, Object... arguments) {
+		InterpreterRuleContext context = new InterpreterRuleContext(parent, invokingState, ruleIndex);
+		if (altIndex > 0) {
+			context.setAltNumber(altIndex);
```

Comment:
💭 Maybe we should make this the caller's responsibility? Right now it's not consistently called.

Reply:
➡️ Now the callers responsibility